kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -869,6 +873,16 @@ func (c *ipamController) allocationIsValid(a *allocation, preferCache bool) bool
 }
 
 func (c *ipamController) syncIPAM() error {
+	ctx := context.Background()
+	ci, err := c.client.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+	if err != nil {
+		err = fmt.Errorf("error getting ClusterInformation: %v", err)
+		return err
+	}
+	if !*ci.Spec.DatastoreReady {
+		return ErrorDatastoreNotReady{}
```

Comment:
Rather than defining a new error type, I think we can simplify this:

```go
if !*ci.Spec.DatastoreReady {
  log.Warn("datastore is locked, skipping ipam sync")
  return nil
}
```

This isn't really an error scenario.

Reply:
Also, we need to check that the pointer is non-nil before we deference it.